pkgs/top-level/all-packages.nix

```diff
@@ -5942,7 +5944,17 @@ with pkgs;
 
   yapf = with python3Packages; toPythonApplication yapf;
 
-  yarn-berry = callPackage ../development/tools/yarn-berry { };
+  inherit
+    ({
+      yarn-berry3 = callPackage ../development/tools/yarn-berry { berryVersion = 3; };
+
+      yarn-berry4 = callPackage ../development/tools/yarn-berry { };
+    })
+    yarn-berry3
+    yarn-berry4
+    ;
```

Comment:
Well, theoretically it could work I suppose. Question is, what is the preferred way? According to https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#recommendation-for-new-packages-with-multiple-versions it seems to be preferred with `inherit`  but I'm fine either way

Reply:
I would prefer what doronbehar wrote, too.

Reply:
> Well, theoretically it could work I suppose. Question is, what is the preferred way? According to [`master`/pkgs/by-name#recommendation-for-new-packages-with-multiple-versions](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name?rgh-link-date=2024-11-12T11%3A33%3A40Z#recommendation-for-new-packages-with-multiple-versions) it seems to be preferred with `inherit` but I'm fine either way

Hmm yes but in this case, there are also these instructions: https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#manual-migration-guidelines which I think are a bit more relevant.